queries/highlights.scm

```diff
@@ -1,132 +1,271 @@
-(comment) @comment
+; Includes
 
 [
-  "("
-  ")"
-  "{"
-  "}"
-  "["
-  "]"
-] @punctuation.bracket
+  "use"
+] @include
 
-[
-  ";"
-  "."
-  ","
-] @punctuation.delimiter
+; Keywords
 
 [
-  "if"
-  "ifdef"
-  "then"
-  "else"
-  "elseif"
-  "end"
-  "try"
-  "while"
-  "for"
-  "use"
-  "as"
-  "var"
-  "let"
-  "embed"
-  "fun"
-  "be"
-  "new"
+  "type"
   "actor"
   "class"
-  "struct"
   "primitive"
   "interface"
   "trait"
-  "type"
+  "struct"
+  "embed"
+  "let"
+  "var"
+  "be"
+  (compile_intrinsic)
+  "as"
+  "consume"
+  "object"
+  "where"
+] @keyword
+
+[
+  "fun"
+] @keyword.function
+
+[
+  "new"
+  "in"
+  "is"
+] @keyword.operator
+
+[
+  "return"
+] @keyword.return
+
+; Qualifiers
+
+[
   "iso"
   "trn"
   "ref"
   "val"
   "box"
   "tag"
+] @type.qualifier
+
+; Conditionals
+
+[
+  "if"
+  "ifdef"
+  "iftype"
+  "then"
+  "else"
+  "elseif"
+  "until"
```

Comment:
i think this should be in the `repeat` section.
It is only used here:

```pony
repeat
   env.out.write(".")
until enough == true end
```
so it should also be marked as such.

Reply:
until precedes a condition though, fitting `@conditional` more imo

Reply:
that is a totally sensible way of thinking about it.